### PR TITLE
Rewrite news feed

### LINKED ISSUE #87: Recent news does not display when returning to the launcher
After returning to the launcher screen from the animator the recent news area does not display. The related error from the console is: `C:\Users\Charlie\Documents\GitHub\animator\app\js\feed.js:7 Uncaught TypeError: undefined is not a function`.

### PR DESCRIPTION
**Pull request by le717**
* Use JSON instead of XML for feed
* Cache news in `localStorage` in case the feed cannot be loaded
* Simplified API

This fixes #87... sorta. While we do switch to a better feed and do all kinds of improvements, the same "`window.api` is not a function" error persists. It seems to be an NW.js issue, as even my workaround (the caching part) suffers from it. :frowning:

After you review and merge, I advise revising the issue with the revised information.

### CODE REVIEW

**Comment by le717** [comment]
- [ ] Implementation TODO: I should probably store the raw post objects instead of the rendered HTML because that makes more sense (IMO) that fully generated HTML.

**Comment by BoatsAreRockable** [comment]
Thanks for your continued work, I will take a look at this tomorrow.

**Comment by le717** [comment]
Alright, change has been made. Review away, but I likely won't be able to fix anything until Monday evening at the earliest.

File: app/js/newsfeed.js [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+                      "Latest news was not cached."
+                    );
+        return true;
+    }
+
+    /**
+     * Retrieve the cached news.
+     *
+     * @returns {Array} An array containing the compiled news,
+     *                  ready for immediate displaying.
```
Comment (le717) [inline app/js/newsfeed.js]: This comment is outdated per the last commit. Will correct with review changes.

File: app/js/newsfeed.js [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+        // Store the news in localStorage in case we cannot fetch it anew
+        window.localStorage.setItem("ba-news", JSON.stringify(posts));
+
+        // Because setItem() does not have a return value upon successful
+        // storage, this kludge (IMO) does that for us.
+        console.info(window.localStorage.getItem("ba-news") ?
+                      "Latest news sucessfully cached." :
+                      "Latest news was not cached."
+                    );
+        return true;
```
Comment (le717) [inline app/js/newsfeed.js]: Maybe we should `return false` if the news could not be cached?

File: app/js/newsfeed.js [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+(function() {
+    "use strict";
+    var newsContainer = window.document.querySelector(".container-news");
+
+    /**
+     * Cache the latest news.
+     *
+     * @param {Array} posts An array containing the compiled news,
+     *                ready for immediate displaying.
```
Comment (le717) [inline app/js/newsfeed.js]: This comment is also outdated. :\

File: app/js/newsfeed.js [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+    </header>
+    <div class="post-content"><p>${post.excerpt}</p></div>
+</div>`;
+    }
+
+    /**
+     * Compile the latest news.
+     *
+     * @param {JSON} data The raw JSON response data.
+     * @returns {Array} The latest news, ready to be displayed.
```
Comment (le717) [inline app/js/newsfeed.js]: Same outdated mess here. :\

**Comment by BoatsAreRockable** [comment]
Sorry about the delay, I've had a look and everything looks in order to me, although admittedly I have not worked with JSON feeds before. Unless you want to add anything more I shall go ahead and merge this.

**Comment by le717** [comment]
I want to fix those outdated comments and get your take on that return value first. ;)

**Comment by le717** [comment]
Final commits pushed, ready for final inspection/merge.

**Comment by le717** [comment]
Reminder: this is ready anytime. :)

**Comment by BoatsAreRockable** [comment]
@le717 Sorry for disappearing off the face of the earth... I will being addressing everything tonight.

**Comment by le717** [comment]
Nah, it's cool. I totally understand. I was just

**Pull request merged**